rma/tests/test_rma.py

```diff
@@ -111,9 +167,70 @@ def _prepare_move(self, product, qty, src, dest):
             'origin': 'Test RMA',
             'location_id': src.id,
             'location_dest_id': dest.id,
+            'picking_id': picking_in.id
         }
         return res
 
+    def test_rma_order_line(self):
+        picking_in = self._create_picking(self.env.ref('base.res_partner_2'))
+        moves_1 = []
+        move_values = self._prepare_move(self.product_1, 3,
+                                         self.stock_location,
+                                         self.customer_location, picking_in)
+        moves_1.append(self.env['stock.move'].create(move_values))
+        wizard_1 = self.rma_add_stock_move.with_context(
+            {'supplier': True,
+             'stock_move_id': [(6, 0, [m.id for m in moves_1])],
+             'active_ids': self.rma_customer_id.id,
+             'active_model': 'rma.order',
+             }
+        ).create({'move_ids': [(6, 0, [m.id for m in moves_1])]})
+        wizard_1.add_lines()
+
+        for line in self.rma_customer_id.rma_line_ids:
+            line.with_context({'default_rma_id': line.rma_id.id
+                               })._default_warehouse_id()
+            line._default_location_id()
+            line.with_context({'partner_id': line.rma_id.partner_id.id
+                               })._default_delivery_address()
+            line._compute_in_shipment_count()
+            line._compute_out_shipment_count()
+            line._compute_procurement_count()
+
+            data = {'reference_move_id': line.reference_move_id.id}
+            new_line = self.rma_line.new(data)
+            new_line._onchange_reference_move_id()
```

Comment:
You should test this more specifically, lines are now the main document (RMA) so it can work by just itself. The workflow is the following (which you should test independently):

1. You start with a done move for a customer. 
2. You create a RMA (rma.order.line).
3. You set the customer and select the move that need to go thorugh a RMA process.
4. At this moment from the user experience all the field are auto-filled (due `_onchange_reference_move_id`). You should assert that everything is filled correctly.

Same workflow should be tested for extension modules:

* rma_sale:  fill customer and the `sale_line_id` and the method `_onchange_sale_line_id` will fill everything.
* rma_account: customer and `invoice_line_id` -> `_onchange_invoice_line_id`.
* rma_purchase (RMA supplier): customer and `purchase_order_line_id` -> `_onchange_purchase_order_line_id`.

Reply:
Okay, So I will test the onchange conditions in all aspects and will ensure that right values are filled into the particular fields.

Is that we have to achieve now, right?